src/main/java/com/ericsson/ei/EndpointSecurity.java

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright 2017 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.ericsson.ei;
+
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.apache.tomcat.util.codec.binary.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@Configuration
+@EnableWebSecurity
+public class EndpointSecurity extends WebSecurityConfigurerAdapter {
+    @Value("${ldap.enabled}")
+    private boolean ldapEnabled;
+    
+    @Value("${ldap.url}")
+    private String ldapUrl;
+    
+    @Value("${ldap.base.dn}")
+    private String ldapBaseDn;
+    
+    @Value("${ldap.username}")
+    private String ldapUsername;
+    
+    @Value("${ldap.password}")
+    private String ldapPassword;
+    
+    @Value("${ldap.user.filter}")
+    private String ldapUserFilter;
+    
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        if(ldapEnabled) {
+            http
+            .authorizeRequests()
+                .anyRequest().authenticated()
+            .and()
+                .sessionManagement()
+                .sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED)
+            .and()
+                .httpBasic()
+            .and()
+                .csrf().disable();
+        }
+        else {
+            http
+            .csrf().disable();
+        }
+    }
```

Comment:
Maybe we should add log a message here which says it configured with LDAP or Not using LDAP. Good to have when debugging in logs.